pkg/kvstore/allocator/allocator.go

```diff
@@ -396,6 +396,48 @@ func (k *kvstoreBackend) Release(ctx context.Context, key allocator.AllocatorKey
 	return nil
 }
 
+// RunLocksGC scans the kvstore for unused locks and removes them
+func (k *kvstoreBackend) RunLocksGC(ctx context.Context, staleKeysPrevRound map[string]kvstore.Value) (map[string]kvstore.Value, error) {
+	// fetch list of all /../locks keys
+	allocated, err := kvstore.ListPrefix(ctx, k.lockPrefix)
+	if err != nil {
+		return nil, fmt.Errorf("list failed: %s", err)
+	}
+
+	staleKeys := map[string]kvstore.Value{}
+
+	// iterate over /../locks
+	for key, v := range allocated {
+		scopedLog := log.WithFields(logrus.Fields{
+			fieldKey:     key,
+			fieldLeaseID: fmt.Sprintf("%x", v.LeaseID),
+		})
+		// Only delete if this key was previously marked as to be deleted
+		if modRev, ok := staleKeysPrevRound[key]; ok &&
+			modRev.ModRevision == v.ModRevision &&
+			modRev.LeaseID == v.LeaseID &&
+			modRev.SessionID == v.SessionID {
+			if err := kvstore.Delete(ctx, key); err != nil {
```

Comment:
Does this time out after a while? Just wondering if this GC somehow got stuck, could we find out from the logs. Presumably `ctx` should bound it but `ctx` is passed as `context.TODO()` so that won't do it for us..